core/arch/arm/include/kernel/thread.h

```diff
@@ -308,6 +323,9 @@ void thread_restore_irq(void);
 #define THREAD_EXCP_ABT	(1 << 2)
 #define THREAD_EXCP_ALL	(THREAD_EXCP_FIQ | THREAD_EXCP_IRQ | THREAD_EXCP_ABT)
 
+/* Defines the bit mask for the foreign interrupts */
+#define THREAD_FOREIGN_INTR	THREAD_EXCP_IRQ
```

Comment:
It's convenient that the bits in the `THREAD_EXCP_*` defines matches corresponding bits in PSTATE etc and I see no reason to change that. But I don't see any reason why you'd ever want to use `THEAD_EXCP_IRQ` directly when you really mean foreign interrupt (or native in the GICv3 case). Every direct use of `THEAD_EXCP_IRQ` would need to be conditional.

We'll need to ifdef `CFG_GICV3` or something in this file to define the macros correctly, so we could just as well do it with bits directly.

By the way, the name of the define should have a proper prefix `THREAD_EXCP_FOREIGN_INTR`.